environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/EnvironmentValidationHandler.java

```diff
@@ -57,20 +73,36 @@ public void accept(Event<EnvironmentDto> environmentDtoEvent) {
                                 validationResult = validationResult.merge(validatorService.validateNetworkWithProvider(environmentDto));
                                 if (validationResult.hasError()) {
                                     goToFailedState(environmentDtoEvent, validationResult.getFormattedErrors());
+                                } else {
+                                    setChildEnvironmentNetworkIfItHasParent(environmentDto);
+                                    environmentDto.getNetwork().setSubnetMetas(cloudNetworkService.retrieveSubnetMetadata(environmentDto,
+                                            environmentDto.getNetwork()));
+                                    environmentResourceService.createAndSetNetwork(environment, environmentDto.getNetwork(), environment.getAccountId(),
+                                            environmentDto.getNetwork().getSubnetMetas());
+                                    environmentService.save(environment);
+                                    goToNetworkCreationState(environmentDtoEvent, environmentDto);
                                 }
-                                goToNetworkCreationState(environmentDtoEvent, environmentDto);
                             } catch (WebApplicationException e) {
                                 String responseMessage = webApplicationExceptionMessageExtractor.getErrorMessage(e);
                                 goToFailedState(environmentDtoEvent, e.getMessage() + ". " + responseMessage);
                             } catch (Exception e) {
                                 goToFailedState(environmentDtoEvent, e.getMessage());
                             }
-                        }, () -> {
-                            goToFailedState(environmentDtoEvent, String.format("Environment was not found with id '%s'.", environmentDto.getId()));
-                        }
+                        }, () -> goToFailedState(environmentDtoEvent, String.format("Environment was not found with id '%s'.", environmentDto.getId()))
                 );
     }
 
+    private void setChildEnvironmentNetworkIfItHasParent(EnvironmentDto currentEnvDto) {
```

Comment:
since from now, we're not setting all the network params (e.g. subnet metas) until this step, and the child's env has to be the same as far as I know as the parent

Reply:
it is not a proper solution, it is a validator, the validator should validate smgt and does not set other things. hard to find later

Reply:
make sense, I'll look for some other approach for this state change stuff